internal/host/host.go

```diff
@@ -312,6 +313,24 @@ func (m *Manager) UpdateInventory(ctx context.Context, h *models.Host, inventory
 	}).Error
 }
 
+func (m *Manager) checkValidationChanged(ctx context.Context, h *models.Host, newValidationRes validationsStatus) (validationsStatus, bool, error) {
+	var currentValidationRes validationsStatus
+	if h.ValidationsInfo != "" {
+		if err := json.Unmarshal([]byte(h.ValidationsInfo), &currentValidationRes); err != nil {
+			return validationsStatus{}, false, errors.Wrapf(err, "Failed to unmarshal validations info from host %s in cluster %s", h.ID, h.ClusterID)
+		}
+	}
+	return currentValidationRes, !reflect.DeepEqual(newValidationRes, currentValidationRes), nil
+}
+
+func (m *Manager) updateValidationsInDB(ctx context.Context, h *models.Host, newValidationRes validationsStatus) (*models.Host, error) {
```

Comment:
This function should get the db as a parameter, in case you are running in a transaction you create a dead lock on the DB.

Reply:
Good catch, updated, thanks!

Reply:
@filanov I told you this happens a lot.

Reply:
1. it's good that we have subsyste,
2. it's easy to test with unit-test, open transaction run the test and commit. i already did it few times.